docs/proposals/pipeline/pipeline.md

```diff
@@ -0,0 +1,415 @@
+	TaskType string `json:"taskType"`
+
+	// Params are key-value pairs of parameters for the predefined task.
+	// These parameters depend on the selected task template.
+	// +optional
+	Params map[string]string `json:"params,omitempty"`
+}
+
+// CustomTask defines the specification for a user-defined task.
+type CustomTask struct {
+	// Image specifies the Docker image name.
+	// More info: https://kubernetes.io/docs/concepts/containers/images
+	// +optional
+	Image string `json:"image,omitempty" protobuf:"bytes,2,opt,name=image"`
+
+	// Command is the entrypoint array. It's not executed in a shell.
+	// If not provided, the image's ENTRYPOINT is used.
+	// Environment variables can be used in the format $(VAR_NAME).
+	// More info: https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#running-a-command-in-a-shell
+	// +optional
+	// +listType=atomic
+	Command []string `json:"command,omitempty" protobuf:"bytes,3,rep,name=command"`
+
+	// Args are the arguments for the entrypoint.
+	// If not provided, the image's CMD is used.
+	// Supports environment variable expansion in the format $(VAR_NAME).
+	// More info: https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#running-a-command-in-a-shell
+	// +optional
+	// +listType=atomic
+	Args []string `json:"args,omitempty" protobuf:"bytes,4,rep,name=args"`
+	
+	// Step's working directory.
+	// If not specified, the container runtime's default will be used, which
+	// might be configured in the container image.
+	// Cannot be updated.
+	// +optional
+	WorkingDir string `json:"workingDir,omitempty" protobuf:"bytes,5,opt,name=workingDir"`
+	
+	// List of environment variables to set in the Step.
+	// Cannot be updated.
+	// +optional
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +listType=atomic
+	Env []corev1.EnvVar `json:"env,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,7,rep,name=env"`
+	
+	// ResourceRequirements required by this Step.
+	// Cannot be updated.
+	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// +optional
+	ResourceRequirements corev1.ResourceRequirements `json:"computeResources,omitempty" protobuf:"bytes,8,opt,name=computeResources"`
+	
+	// Script is the contents of an executable file to execute.
+	// If Script is not empty, the CustomTask cannot have a Command and the Args will be passed to the Script.
+	// +optional
+	Script string `json:"script,omitempty"`
+}
+
+type PipelineStatus struct {
+	// Phase describes the overall state of the Pipeline.
+	// +optional
+	Phase *string `json:"phase,omitempty"`
+
+    // EventListenerServiceName specifies the name of the service created by Kurator for event listeners. 
+    // This name is useful for users when setting up a gateway service and routing to this service.
+    // +optional
+    EventListenerServiceName *string `json:"eventListenerServiceName,omitempty"`
```

Comment:
indent

Reply:
I'm not sure what the 'indent' means here

Reply:
fixed